❌ **Build Blocked:** Step 1/1 (`Address review feedback`) was **not committed** because build errors could not be resolved after 3 fix attempts + full code regeneration.

This step needs manual review or will be addressed in a follow-up.

<details>
<summary>Build Errors (last attempt)</summary>

```
C:\Agents\principalengineer-91ee385192b64cff9e2ac751d2daea99\ReportingDashboard\tests\ReportingDashboard.Tests\Integration\DashboardComponentIntegrationTests.cs(32,23): error CS1729: 'DashboardDataService' does not contain a constructor that takes 1 arguments [C:\Agents\principalengineer-91ee385192b64cff9e2ac751d2daea99\ReportingDashboard\tests\ReportingDashboard.Tests\ReportingDashboard.Tests.csproj]
C:\Agents\principalengineer-91ee385192b64cff9e2ac751d2daea99\ReportingDashboard\tests\ReportingDashboard.Tests\Unit\Services\DashboardDataServiceValidationTests.cs(54,53): error CS1729: 'DashboardDataService' does not contain a constructor that takes 1 arguments [C:\Agents\principalengineer-91ee385192b64cff9e2ac751d2daea99\ReportingDashboard\tests\ReportingDashboard.Tests\ReportingDashboard.Tests.csproj]
C:\Agents\principalengineer-91ee385192b64cff9e2ac751d2daea99\ReportingDashboard\tests\ReportingDashboard.Tests\Integration\DashboardComponentIntegrationTests.cs(39,23): error CS1729: 'DashboardDataService' does not contain a constructor that takes 1 arguments [C:\Agents\principalengineer-91ee385192b64cff9e2ac751d2daea99\ReportingDashboard\tests\ReportingDashboard.Tests\ReportingDashboard.Tests.csproj]
C:\Agents\principalengineer-91ee385192b64cff9e2ac751d2daea99\ReportingDashboard\tests\ReportingDashboard.Tests\Unit\Components\DashboardPageTests.cs(19,23): error CS1729: 'DashboardDataService' does not contain a constructor that takes 1 arguments [C:\Agents\principalengineer-91ee385192b64cff9e2ac751d2daea99\ReportingDashboard\tests\ReportingDashboard.Tests\ReportingDashboard.Tests.csproj]
C:\Agents\principalengineer-91ee385192b64cff9e2ac751d2daea99\ReportingDashboard\tests\ReportingDashboard.Tests\Unit\Components\DashboardPageTests.cs(66,23): error CS1729: 'DashboardDataService' does not contain a constructor that takes 1 arguments [C:\Agents\principalengineer-91ee385192b64cff9e2ac751d2daea99\ReportingDashboard\tests\ReportingDashboard.Tests\ReportingDashboard.Tests.csproj]
C:\Agents\principalengineer-91ee385192b64cff9e2ac751d2daea99\ReportingDashboard\tests\ReportingDashboard.Tests\Unit\Components\TimelineTests.cs(33,35): error CS0246: The type or namespace name 'Timeline' could not be found (are you missing a using directive or an assembly reference?) [C:\Agents\principalengineer-91ee385192b64cff9e2ac751d2daea99\ReportingDashboard\tests\ReportingDashboard.Tests\ReportingDashboard.Tests.csproj]
C:\Agents\principalengineer-91ee385192b64cff9e2ac751d2daea99\ReportingDashboard\tests\ReportingDashboard.Tests\Unit\Components\TimelineTests.cs(34,40): error CS1503: Argument 2: cannot convert from 'ReportingDashboard.Models.TimelineData' to 'string' [C:\Agents\principalengineer-91ee385192b64cff9e2ac751d2daea99\ReportingDashboard\tests\ReportingDashboard.Tests\ReportingDashboard.Tests.csproj]
C:\Agents\principalengineer-91ee385192b64cff9e2ac751d2daea99\ReportingDashboard\tests\ReportingDashboard.Tests\Integration\DashboardDataServiceIntegrationTests.cs(43,23):...
```
</details>

--- a/src/ReportingDashboard/Components/Heatmap.razor
+++ b/src/ReportingDashboard/Components/Heatmap.razor
@@ -1,29 +1,58 @@
-<div class="hm-title">MONTHLY EXECUTION HEATMAP – SHIPPED · IN PROGRESS · CARRYOVER · BLOCKERS</div>
-<div class="hm-grid" style="display:grid; grid-template-columns:@GridColumns; grid-template-rows:@GridRows; border:1px solid #E0E0E0; flex:1; min-height:0;">
-    <div class="hm-corner">STATUS</div>
-    @foreach (var month in Months)
-    {
-        var isCurrentMonth = month.Equals(CurrentMonth, StringComparison.OrdinalIgnoreCase);
-        <div class="hm-col-hdr @(isCurrentMonth ? "apr-hdr" : "")">
-            @month @(isCurrentMonth ? "▸ Now" : "")
-        </div>
-    }
+@if (Months is not null && Months.Count > 0)
+{
+    <div class="hm-wrap">
+        <div class="hm-title">MONTHLY EXECUTION HEATMAP - SHIPPED &middot; IN PROGRESS &middot; CARRYOVER &middot; BLOCKERS</div>
+        <div class="hm-grid" style="grid-template-columns: @GridColumns; grid-template-rows: @GridRows;">
+            <div class="hm-corner">STATUS</div>
+            @foreach (var month in Months)
+            {
+                var headerClass = IsCurrentMonth(month) ? "hm-col-hdr apr-hdr" : "hm-col-hdr";
+                <div class="@headerClass">@month</div>
+            }
 
-    <HeatmapRow CategoryLabel="✓ SHIPPED" CssPrefix="ship"
-                Items="@HeatmapData.Shipped" Months="@Months" CurrentMonth="@CurrentMonth" />
-    <HeatmapRow CategoryLabel="▶ IN PROGRESS" CssPrefix="prog"
-                Items="@HeatmapData.InProgressItems" Months="@Months" CurrentMonth="@CurrentMonth" />
-    <HeatmapRow CategoryLabel="⟳ CARRYOVER" CssPrefix="carry"
-                Items="@HeatmapData.Carryover" Months="@Months" CurrentMonth="@CurrentMonth" />
-    <HeatmapRow CategoryLabel="⚑ BLOCKERS" CssPrefix="block"
-                Items="@HeatmapData.Blockers" Months="@Months" CurrentMonth="@CurrentMonth" />
-</div>
+            @if (HeatmapData is not null)
+            {
+                <HeatmapRow CategoryLabel="&#x2705; SHIPPED" CategoryKey="shipped" CssPrefix="ship" Items="@(HeatmapData.Shipped ?? new Dictionary<string, List<string>>())" Months="@Months" CurrentMonth="@CurrentMonth" />
+                <HeatmapRow CategoryLabel="&#x1F504; IN PROGRESS" CategoryKey="inProgress" CssPrefix="prog" Items="@GetItemsByPropertyIndex(1)" Months="@Months" CurrentMonth="@CurrentMonth" />
+                <HeatmapRow CategoryLabel="&#x1F4CB; CARRYOVER" CategoryKey="carryover" CssPrefix="carry" Items="@(HeatmapData.Carryover ?? new Dictionary<string, List<string>>())" Months="@Months" CurrentMonth="@CurrentMonth" />
+                <HeatmapRow CategoryLabel="&#x1F6AB; BLOCKERS" CategoryKey="blockers" CssPrefix="block" Items="@(HeatmapData.Blockers ?? new Dictionary<string, List<string>>())" Months="@Months" CurrentMonth="@CurrentMonth" />
+            }
+        </div>
+    </div>
+}
 
 @code {
-    [Parameter] public HeatmapData HeatmapData { get; set; } = default!;
-    [Parameter] public List<string> Months { get; set; } = new();
-    [Parameter] public string CurrentMonth { get; set; } = string.Empty;
+    [Parameter]
+    public HeatmapData? HeatmapData { get; set; }
 
-    private string GridColumns => $"160px repeat({Months.Count}, 1fr)";
+    [Parameter]
+    public List<string>? Months { get; set; }
+
+    [Parameter]
+    public string CurrentMonth { get; set; } = string.Empty;
+
+    private string GridColumns => $"160px repeat({Months?.Count ?? 0}, 1fr)";
+
     private string GridRows => "36px repeat(4, 1fr)";
+
+    private bool IsCurrentMonth(string month) =>
+        month.Equals(CurrentMonth, StringComparison.OrdinalIgnoreCase);
+
+    private Dictionary<string, List<string>> GetItemsByPropertyIndex(int index)
+    {
+        if (HeatmapData is null)
+            return new Dictionary<string, List<string>>();
+
+        var props = HeatmapData.GetType().GetProperties()
+            .Where(p => p.PropertyType == typeof(Dictionary<string, List<string>>))
+            .ToArray();
+
+        if (index < props.Length)
+        {
+            return props[index].GetValue(HeatmapData) as Dictionary<string, List<string>>
+                ?? new Dictionary<string, List<string>>();
+        }
+
+        return new Dictionary<string, List<string>>();
+    }
 }

--- a/tests/ReportingDashboard.Tests/ReportingDashboard.Tests.csproj
+++ b/tests/ReportingDashboard.Tests/ReportingDashboard.Tests.csproj
@@ -1,0 +1,8 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>net8.0</TargetFramework>
+    <Nullable>enable</Nullable>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <IsPackable>false</IsPackable>
+  </PropertyGroup>
+</Project>

--- a/tests/ReportingDashboard.UITests/ReportingDashboard.UITests.csproj
+++ b/tests/ReportingDashboard.UITests/ReportingDashboard.UITests.csproj
@@ -1,0 +1,8 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>net8.0</TargetFramework>
+    <Nullable>enable</Nullable>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <IsPackable>false</IsPackable>
+  </PropertyGroup>
+</Project>

--- a/tests/UITests/Project.UITests.csproj
+++ b/tests/UITests/Project.UITests.csproj
@@ -1,0 +1,8 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>net8.0</TargetFramework>
+    <Nullable>enable</Nullable>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <IsPackable>false</IsPackable>
+  </PropertyGroup>
+</Project>

